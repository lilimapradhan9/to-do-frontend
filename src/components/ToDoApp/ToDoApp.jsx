import React, {Component} from "react";

export default class ToDoApp extends Component {
    constructor(props) {
        super(props);
        this.state = {todo: ''};
    }

    myChangeHandler = (event) => {
        this.setState(
            {todo: event.target.value}
        );
    }

    render() {
        return (<>
                <form>
                    <h1>Hellooo</h1>
                    <p>Enter Todo</p>
                    <input
                        aria-label={"todo"}
                        type='text'
                        onChange={this.myChangeHandler}
                    />
                </form>
                {
                    this.state.todo &&
                    <>
                        <h1>Todo List</h1>
                        <h6>{this.state.todo}</h6>
                    </>
                }
            </>
        );
    }
}