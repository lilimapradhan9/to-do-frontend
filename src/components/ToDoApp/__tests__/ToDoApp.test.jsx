import ToDoApp from "../ToDoApp";
import React from "react";
import {render, fireEvent} from "@testing-library/react";

describe('To do App', () => {
    it('should load todo input form', async () => {
        const todoApp = render(<ToDoApp/>);

        expect(todoApp.baseElement).toMatchSnapshot();
    })

    it('should should display todo', async () => {
        const todoApp = render(<ToDoApp/>);

        const inputElement = todoApp.getByLabelText("todo");
        fireEvent.change(inputElement, { target: { value: 'abcd' } })

        expect(todoApp.baseElement).toMatchSnapshot();
    })
});