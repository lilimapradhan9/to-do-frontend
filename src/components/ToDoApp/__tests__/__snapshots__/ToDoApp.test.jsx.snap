// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To do App should load todo input form 1`] = `
<body>
  <div>
    <form>
      <h1>
        Hellooo
      </h1>
      <p>
        Enter Todo
      </p>
      <input
        aria-label="todo"
        type="text"
      />
    </form>
    
  </div>
</body>
`;

exports[`To do App should should display todo 1`] = `
<body>
  <div>
    <form>
      <h1>
        Hellooo
      </h1>
      <p>
        Enter Todo
      </p>
      <input
        aria-label="todo"
        type="text"
      />
    </form>
    <h1>
      Todo List
    </h1>
    <h6>
      abcd
    </h6>
  </div>
</body>
`;
